### PR TITLE
Explain page: handle decision issues with non-Appeal decision reviews

### PR DESCRIPTION
**Pull request by yoomlam**
Addendum to PR #16409

### Description
For the appeal narrative of the `explain` page, don't fail if there are `DecisionIssues` associated with non-`Appeal` decision reviews, such as HLRs and SCs.

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] `DecisionIssues` associated with non-`Appeal` decision reviews do not cause the `explain` page to not load
- [ ] No PII in JSON file

### Testing Plan
See new RSpec test.

### CODE REVIEW

File: app/controllers/concerns/explain_appeal_events_concern.rb [hunk 1]
```diff
@@ -55,18 +55,25 @@ def request_issues_as_event_data
     end
 
     # Remove records after processing them.
-    # Don't remove them while processing because multiple request_issues can map to same decision_issue.
+    # Don't remove them while processing because multiple request_issues can map to the same decision_issue.
     exported_records(RequestDecisionIssue).each do |req_dec_issue|
       req_issues.delete(req_dec_issue["request_issue_id"])
       dec_issues.delete(req_dec_issue["decision_issue_id"])
     end
-    fail "Remaining DecisionIssue are not associated: #{dec_issues}" unless dec_issues.blank?
```
Comment (yoomlam) [inline app/controllers/concerns/explain_appeal_events_concern.rb:63]: Don't fail completely. Just log a warning -- line 75.

File: spec/feature/explain_spec.rb [hunk 2]
```diff
@@ -126,5 +126,20 @@
       task = real_appeal.tasks.sample
       expect(page).to have_content("#{task.type}_#{task.id}")
     end
+
+    context "appeal with request issue contesting an HLR decision" do
+      let(:json_filename) { "appeal-106435.json" }
```
Comment (yoomlam) [inline spec/feature/explain_spec.rb:131]: Sanitized JSON for an appeal in prod -- https://github.com/department-of-veterans-affairs/caseflow/wiki/Exporting-and-Importing-Appeals#via-rails-console

File: spec/feature/explain_spec.rb [hunk 3]
```diff
@@ -126,5 +126,20 @@
...
+      before do
+        # 106435.json requires this HLR to exist because a decision_issue references it
+        FactoryBot.create(:higher_level_review, id: 2_000_085_625)
```
Comment (yoomlam) [inline spec/feature/explain_spec.rb:134]: The JSON Exporter doesn't export HLR or SC currently, so we just create them based on the HLR's id in the JSON file.

**Review by lowellrex — APPROVED** [review]
LGTM!

**Pull request merged**